app/Exceptions/Handler.php

```diff
@@ -108,9 +113,24 @@ public function render($request, Exception $e)
                 }
             }
 
-            return response()->json([
+            $response = response()->json([
                 'errors' => $errors
             ], $statusCode, $headers);
+
+            // In the circumstance where an exception is raised
+            // before the lumen request cycle reaches the middleware stage,
+            // it is necessary to force the inclusion of the CORS headers.
+
+            // This uses the app's CORS config.
+            // This config must be loaded in the bootstrap process
+            // before exception handlers are expected to be used.
+            $options = config('cors');
+
+            // This CorsService relies on Asm89\Stack
+            $cors = new CorsService($options);
+            $response = $cors->addActualRequestHeaders($response, $request);
```

Comment:
We could add a check here to see if the headers are already added.

Reply:
@willdoran Do you mean execute addActualRequestHeaders and then check to see if it worked? Or do you mean to check if they're already there and only run if they're not?

Reply:
Sorry to be more clear, I think we can do something like
if (! $response->headers->has('Access-Control-Allow-Origin')) {
            $response = $cors->addActualRequestHeaders($response, $request);
        }

Reply:
@crcommons ^

Reply:
Thanks! I couldn't tell what you meant from the location of the comment.